volatility3/framework/plugins/windows/modules.py

```diff
@@ -50,49 +59,74 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
             ),
         ]
 
+    def dump_module(self, session_layers, pe_table_name, mod):
+        session_layer_name = self.find_session_layer(
+            self.context, session_layers, mod.DllBase
+        )
+        file_output = f"Cannot find a viable session layer for {mod.DllBase:#x}"
+        if session_layer_name:
+            file_handle = dlllist.DllList.dump_pe(
+                self.context,
+                pe_table_name,
+                mod,
+                self.open,
+                layer_name=session_layer_name,
+            )
+            file_output = "Error outputting file"
+            if file_handle:
+                file_output = file_handle.preferred_filename
+
+        return file_output
+
+    def process_module(self, session_layers, pe_table_name, mod):
+        if self.config["base"] and self.config["base"] != mod.DllBase:
+            return None
+
+        try:
+            BaseDllName = mod.BaseDllName.get_string()
+        except exceptions.InvalidAddressException:
+            BaseDllName = ""
+
+        if self.config["name"] and self.config["name"] not in BaseDllName:
+            return None
+
+        try:
+            FullDllName = mod.FullDllName.get_string()
+        except exceptions.InvalidAddressException:
+            FullDllName = ""
+
+        file_output = "Disabled"
+        if self.config["dump"]:
+            file_output = self.dump_module(session_layers, pe_table_name, mod)
```

Comment:
If this is just using `pe_table_name` and `session_layers` to pass through to the `dump_module` call, then this should do what it's supposed to do, and the person calling it should call `dump_module` directly instead themselves.